rust/private/rustc.bzl

```diff
@@ -993,6 +993,11 @@ def _create_extra_input_args(ctx, file, build_info, dep_info):
         input_files.append(build_info.out_dir)
         input_files.append(build_info.link_flags)
 
+    # Populate with transitive link search paths
+    for dep_build_info in dep_info.transitive_build_infos.to_list():
```

Comment:
So this code here is quadratic in extreme situations (and there is pre-existing code that has the same problem, I'm just trying to improve things incrementally). I'd prefer if there was a depset of files field in DepInfo directly, and the resulting code here would look like:

```
build_flags_files.append(dep_info.link_search_path_files)
input_files.append(dep_info.link_search_path_files)
```

Reply:
I took your advice in the recent commit. I added a new `link_search_path_files` field to DepInfo and now flatten them to propagate.